src/envoy/auth/jwt.cc

```diff
@@ -139,75 +182,155 @@ bool VerifySignature(bssl::UniquePtr<EVP_PKEY> key, const std::string &alg,
   return true;
 }
 
-bool VerifySignature(const std::string &pkey_pem, const std::string &alg,
-                     const std::string &signature,
+bool VerifySignature(const bssl::UniquePtr<EVP_PKEY> &key,
+                     const std::string &alg, const std::string &signature,
                      const std::string &signed_data) {
-  return VerifySignature(EvpPkeyFromStr(pkey_pem), alg, CastToUChar(signature),
-                         signature.length(), CastToUChar(signed_data),
-                         signed_data.length());
+  return VerifySignature(key, alg, CastToUChar(signature), signature.length(),
+                         CastToUChar(signed_data), signed_data.length());
 }
 
 }  // namespace
 
 namespace Jwt {
+namespace {
+
+class Internal {
```

Comment:
Please have a meaningful class name.
Also add class comment to describe its purpose